### PR TITLE
setup: fix version identification

### PR DESCRIPTION
**Pull request by oliver-sanders**
In the `setup.py` we have been determining the version via:

```python
from metomi.isodatetime import __version__
```

Which makes good sense, but, looking back at it again it seems a bit dodge, if we already have `isodatetime` installed will it not pick up that version then install the current code under that version?

Solution, import the `__init__.py` file by relative path.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
Tests break.

**Comment by kinow** [comment]
Huh, one build in Travis failed with

```
coverage run setup.py test --pytest-args='--runslow' &
TEST_PID="$!"
minutes=0
limit=50
while ps -p "${TEST_PID}" >/dev/null 2>&1; do
  echo -n -e " \b" # never leave evidences!
  if ((minutes == limit)); then
    exit 1
  fi
  minutes=$((minutes + 1))
  sleep 60
done
wait "${TEST_PID}"
Traceback (most recent call last):
  File "setup.py", line 48, in <module>
    Path('metomi/isodatetime/__init__.py')
  File "setup.py", line 40, in get_version
    spec = spec_from_file_location(module, path)
  File "<frozen importlib._bootstrap_external>", line 581, in spec_from_file_location
AttributeError: 'PosixPath' object has no attribute 'endswith'
Coverage.py warning: Module ./isodatetime was never imported. (module-not-imported)
Coverage.py warning: No data was collected. (no-data-collected)
The command "export COVERAGE_PROCESS_START="${TRAVIS_BUILD_DIR}/.coveragerc"
```

So just kicked it to see it that does the trick.

File: setup.py [hunk 1]
```diff
@@ -16,16 +16,37 @@
...
+    """Return the __version__ attr from a module sourced by FS path."""
+    spec = spec_from_file_location(module, path)
+    module = module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module.__version__
+
+
+__version__ = get_version(
+    'metomi.isodatetime',
+    Path('metomi/isodatetime/__init__.py')
```
Comment (kinow) [inline setup.py]: Looks like it doesn't work for Python 3.5:

```
Traceback (most recent call last):
  File "setup.py", line 48, in <module>
    Path('metomi/isodatetime/__init__.py')
  File "setup.py", line 40, in get_version
    spec = spec_from_file_location(module, path)
  File "<frozen importlib._bootstrap_external>", line 581, in spec_from_file_location
AttributeError: 'PosixPath' object has no attribute 'endswith'
```

Comment (oliver-sanders) [reply setup.py]: Hmmm? It works for me on Python3.4 (don't have 3.5 at hand):

```console
$ python3.4
Python 3.4.10 (default, Apr  8 2019, 02:17:36) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-36)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pathlib import Path
>>> Path('metomi/isodatetime/__init__.py')
PosixPath('metomi/isodatetime/__init__.py')
```

Comment (oliver-sanders) [reply setup.py]: Ahhh, got it, the `spec_from_file_location` function only picked up Pathlib support at 3.6!

https://docs.python.org/3/library/importlib.html#importlib.util.spec_from_file_location

Comment (oliver-sanders) [reply setup.py]: Have added Python3.5 support as it was simple to do.

File: setup.py [hunk 2]
```diff
@@ -16,16 +16,37 @@
...
 # https://docs.pytest.org/en/2.8.7/goodpractices.html#manual-integration
 from setuptools.command.test import test as TestCommand
 from setuptools import setup, find_namespace_packages
+import sys
+
+
+try:
+    os.chdir(os.path.dirname(__file__))
+except FileNotFoundError:
+    pass
```
Comment (matthewrmshin) [inline setup.py]: Currently, if the current working directory is not the same directory containing the library, the default behaviour for something like `python3 /path/to/metomi/isodatetime/setup.py bdist` is to build a distribution file under `$PWD/dist/`.

Does this change the behaviour? (May surprise some usage.)

Comment (matthewrmshin) [reply setup.py]: @oliver-sanders This block is still changing the current working directory after your latest commit. Is this still necessary?

Comment (oliver-sanders) [reply setup.py]: Forgot to commit the change which removed it.

**Pull request merged**